cpp/include/raft/neighbors/cagra_hnswlib_types.hpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "ann_types.hpp"
+#include <memory>
+#include <raft/distance/distance_types.hpp>
+
+#include <cstdint>
+#include <sys/types.h>
+#include <type_traits>
+
+namespace raft::neighbors::cagra_hnswlib {
```

Comment:
Rather than a separate index, namespace, and file for cagra_hnsw, can't we just use the cagra index, provide a helper in raft::neighbors::hnsw to convert it to the HNSW index wrapper, and then provide a raft::neighbors::hnsw::search()? I understand using the cagra_HNSW algo name for the benchmarks but I think HNSW itself is more appropriate for the actual public api to do this.